File: Unidad 5/c#/Programa_5_Unidad_5.cs
using System;

namespace Programa_5_Unidad_5 // Note: actual namespace depends on the project name.
{
    internal class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Ingrese una frase o palabra:");
            string entrada_Usuario = Console.ReadLine();

            int cantidad_Palabras = Contar_Palabras(entrada_Usuario);

            Console.WriteLine("Número de palabras: " + cantidad_Palabras);
        }

        static int Contar_Palabras(string cadena)
        {
            if (string.IsNullOrEmpty(cadena))
            {
                return 0;
            }

            string[] palabras = cadena.Split(new char[] { ' ', '\t', '\n', '\r' }, StringSplitOptions.RemoveEmptyEntries);

            return palabras.Length;
        }
    }
}

File: Unidad 5/Java/Programa_5_Unidad_5.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package com.mycompany.programa_5_unidad_5;
import java.util.Scanner;
/**
 *
 * @author USUARIO LENOVO
 */
public class Programa_5_Unidad_5 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Ingrese una frase o palabra:");
        String entrada_Usuario = scanner.nextLine();

        int cantidad_Palabras = contar_Palabras(entrada_Usuario);

        System.out.println("Número de palabras: " + cantidad_Palabras);
    }

    public static int contar_Palabras(String cadena) {
        if (cadena == null || cadena.isEmpty()) {
            return 0;
        }

        String[] palabras = cadena.split("\\s+");

        return palabras.length;
    }
}
